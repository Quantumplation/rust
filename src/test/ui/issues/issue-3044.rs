fn main() {
    let needlesArr: Vec<char> = vec!['a', 'f'];
    needlesArr.iter().fold(|x, y| {
    });
    //~^^ ERROR mismatched types
    //~| ERROR arguments to this function are incorrect
}
