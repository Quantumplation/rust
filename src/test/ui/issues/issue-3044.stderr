error[E0308]: mismatched types
  --> $DIR/issue-3044.rs:3:35
   |
LL |       needlesArr.iter().fold(|x, y| {
   |  ___________________________________^
LL | |     });
   | |_____^ expected closure, found `()`
   |
   = note: expected closure `[closure@$DIR/issue-3044.rs:3:28: 4:6]`
            found unit type `()`

error[E0308]: arguments to this function are incorrect
  --> $DIR/issue-3044.rs:3:23
   |
LL |     needlesArr.iter().fold(|x, y| {
   |                       ^^^^ an argument of type _ is missing
   |
help: provide the argument
   |
LL |     needlesArr.iter().(|x, y| {
LL |     }, {_})(|x, y| {
   |

error: aborting due to 2 previous errors

For more information about this error, try `rustc --explain E0308`.
